src/wallet/rpcwallet.cpp

```diff
@@ -3297,16 +3297,18 @@ UniValue rescanblockchain(const JSONRPCRequest& request)
         }
     }
 
-    const CBlockIndex* stopBlock = pwallet->ScanForWalletTransactions(pindexStart, pindexStop, reserver, true);
-    if (!stopBlock) {
-        if (pwallet->IsAbortingRescan()) {
-            throw JSONRPCError(RPC_MISC_ERROR, "Rescan aborted.");
-        }
-        // if we got a nullptr returned, ScanForWalletTransactions did rescan up to the requested stopindex
+    const CBlockIndex* stopBlock = nullptr;
+    CWallet::ScanResult result =
+        pwallet->ScanForWalletTransactions(pindexStart, pindexStop, reserver, stopBlock, true);
+    switch (result) {
+    case CWallet::ScanResult::SUCCESS:
         stopBlock = pindexStop ? pindexStop : pChainTip;
```

Comment:
Why are you overriding the return value?

Reply:
This has been the behavior since the rpc's introduction in c77170fbdb

Reply:
I can pick this up and address my feedback middle of next week. (Let me know if you are currently busy or want to take a stab at this.)